packages/frontend/src/Build.svelte

```diff
@@ -258,6 +275,47 @@ async function updateAvailableArchitectures(selectedImage: string) {
   }
 }
 
+// Updates the filesystem selection based upon the select image,
+// specifically if the image is Fedora we will  have to select the filesystem (it cannot be default).
+async function detectFedoraImageFilesystem(selectedImage: string) {
+  const image = findImage(selectedImage);
+  let imageLabels = image?.Labels;
+
+  // If it is a manifest, we must find the child images associated to the manifest
+  // in order to get the labels used to determine if it's based on Fedora or not
+  if (image?.isManifest) {
+    try {
+      const manifest = await bootcClient.inspectManifest(image);
+      const foundImages = await findImagesAssociatedToManifest(manifest);
+
+      // Just get the labels from the first image, as they should all be the same.
+      imageLabels = foundImages[0].Labels;
+    } catch (error) {
+      console.error('Error inspecting manifest:', error);
+    }
+  }
+
+  // We use a specical label to determine what the bootc image was built against. We do not use "annotations" due to limitations of the PD API
+  // that does not show the annotations of the inspect image.
+  // Each build will have 'ostree.linux' added as a label, within that label it is either el9 (for centos / rhel) or fc40 for Fedora.
+  // the format for example will be: "ostree.linux": "5.14.0-437.el9.x86_64", or "ostree.linux": "6.8.9-300.fc40.aarch64",
+  // we can use this to determine if the bootc image was built upon.
+  // We will use regex to determine if it is fedora or not by checking if it contains "fcNUMBER" where NUMBER is the version of Fedora.
+  if (imageLabels && imageLabels['ostree.linux']) {
+    const label = imageLabels['ostree.linux'];
+    if (label.match(/fc\d+/)) {
```

Comment:
This makes me cringe a little, but I know we don't have many options and should be safe enough.  :)
Maybe we should match the '.' first too?

Reply:
Yeah :( It makes me cringe too, but unfortunately annotations is not available via the API (at the moment) and luckily ostree.linux **does** say if it's centos or fedora.